alf/trainers/policy_trainer.py

```diff
@@ -319,13 +319,14 @@ def play(root_dir,
     episode_length = 0
     episodes = 0
     while episodes < num_episodes:
-        time_step, policy_state = _step(
-            algorithm=algorithm,
-            env=env,
-            time_step=time_step,
-            policy_state=policy_state,
-            epsilon_greedy=epsilon_greedy,
-            metrics=[])
+        with torch.no_grad():
+            time_step, policy_state = _step(
+                algorithm=algorithm,
```

Comment:
Since batch-norm might be used in algorithm, I think here we may need to turn on the ```eval``` mode by using ```algorithm.eval()```, which will freeze the batch-norm statistics. 
We might can also do the similar thing for the periodic evaluation during training, but this might not be very critical and can be done in the future.

Reply:
I am aware of this. We can do it in the future.

Reply:
And it seems that pytorch handles batch_size=1 correctly, since it can correctly play even without using eval()

Reply:
Need to check the source code to confirm though

Reply:
I think during play, the initial running mean and std of batch-norm are constructed by loading from checkpoint, therefore it can run without error for batch_size =1. But without eval, the statistics are still changing.
Also for training, I guess if we set ```num_parallel_environments=1``` there will be problem. We can do more investigation in the future.